### PR TITLE
Use Mergify for merge-when-green

### PR DESCRIPTION
**Pull request by fleupold**
Merge when green does not support automatically updating the PR branch if it is out of date with master. When multiple people submit PRs at around the same time this can require multiple manual visits of the PR page to update against current master.

[Mergify](https://doc.mergify.io/index.html) seems to be a much more powerful tool that can also allow us to do things such as automatic rebases if the base branch gets merged, auto-merge dependabot changes etc.

This PR aims at just replacing the basic functionality we have today (merge when green) with mergify. This should also work on outdated branches.

### Test Plan
Wait for approval. Then push empty commit into master (to make it out of date) and attach merge-when-green label expect PR to be automatically merged after rebasing and running CI again.

### CODE REVIEW

**Review by e00E — APPROVED** [review]
I don't know either merge tool but we can certainly try a new one.

**Comment by fleupold** [comment]
Note, that for us nothing should change. We still tag a PR we want to see automatically merged with the `merge when green` label. The new tool not only wait for CI to pass in that case but also should automatically update the base branch before merging.

**Comment by fleupold** [comment]
> For security reason, Mergify will never automatically merge a pull request if it changes the Mergify configuration file. 
(https://doc.mergify.io/getting-started.html)

Guess I have to manually merge this one and test with another PR

**Pull request merged**
